compiler-rt/lib/scudo/standalone/combined.h

```diff
@@ -565,6 +565,19 @@ class Allocator {
             storeSecondaryAllocationStackMaybe(Options, OldPtr, NewSize);
           }
         }
+
+        // If we've increased the size, fill the extra bytes.
+        if (NewSize > OldSize) {
```

Comment:
From the rest of the code, you need to do this for allocations coming from the secondary, but allocations coming from the primary have the entire size filled on allocation (see initChunk). The only way that a primary allocation would need this is if the allocation shrunk and then expanded. That operation is very, very unlikely to happen. But to make this still work in that case, you could modify this so that if a primary allocation shrinks, then set all of the bytes after the shrunk allocation to the fill. That would mean the performance of the normal case doesn't require this extra memset for primary allocations.

Reply:
I agree with the potential performance cost here. Note that both primary and secondary have the entire block filled (secondary prepares block with content filling, correct me if I'm wrong). And I like the solution, we do content filling only when it shrinks (don't need to check if it's primary or secondary)

Reply:
Done. I like this idea too, thanks! I agree that with both allocators the entire block is filled at allocation time (in initChunk for the primary, or in the allocator itself for the secondary [except if it just got fresh known-to-be-zero memory from the OS]) and this change maintains this invariant for the unused portion of each block